tiledb/sm/storage_manager/storage_manager.h

```diff
@@ -144,23 +144,29 @@ class StorageManager {
    * Opens an array for reads at a timestamp. All the metadata of the
    * fragments created before or at the input timestamp are retrieved.
    *
+   * If a timestamp_start is provided, this API will open the array between
+   * `timestamp_start` and `timestamp_end`.
+   *
    * @param array_uri The array URI.
-   * @param timestamp The timestamp at which the array will be opened.
-   *     In TileDB, timestamps are in ms elapsed since
-   *     1970-01-01 00:00:00 +0000 (UTC).
    * @param enc_key The encryption key to use.
    * @param array_schema The array schema to be retrieved after the
    *     array is opened.
    * @param fragment_metadata The fragment metadata to be retrieved
    *     after the array is opened.
+   * @param timestamp_end The timestamp at which the array will be opened.
+   *     In TileDB, timestamps are in ms elapsed since
+   *     1970-01-01 00:00:00 +0000 (UTC).
+   * @param timestamp_start The (optional) starting timestamp to open the array
+   * between, starting at this timestamp and ending at `timestamp_end`.
    * @return Status
    */
   Status array_open_for_reads(
       const URI& array_uri,
-      uint64_t timestamp,
       const EncryptionKey& enc_key,
       ArraySchema** array_schema,
-      std::vector<FragmentMetadata*>* fragment_metadata);
+      std::vector<FragmentMetadata*>* fragment_metadata,
+      uint64_t timestamp_end,
```

Comment:
As mentioned above, let's switch the order of start and end. It will be less error-prone in the future.